controllers/flinkcluster_reconciler.go

```diff
@@ -213,11 +214,23 @@ func (reconciler *_ClusterReconciler) reconcileJob() error {
 	var log = reconciler.log
 	var desiredJob = reconciler.desiredState.job
 	var observedJob = reconciler.observedState.job
-
-	if desiredJob != nil && observedJob == nil {
-		return reconciler.createJob(desiredJob)
+	var observedClusterComponents = reconciler.observedState.cluster.Status.Components
+	if desiredJob != nil {
+		if observedJob == nil {
```

Comment:
how's this possible under the if condition above?

Reply:
This is a normal case, "desiredJob != nil" means we want to create a job, while "observedJob == nil" means we haven't created yet, then the action will be just to create one.